### PR TITLE
Rob bagby extension support

### PR DESCRIPTION
**Pull request by JackQuincy**
**What this PR does / why we need it**:
This enables custom extensions that can be shared with other devs. 
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
#599
This is just a restart on this pr
#687

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1304)

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -98,7 +98,7 @@ imports:
   version: 5bf94b69c6b68ee1b541973bb8e1144db23a194b
 - name: github.com/Sirupsen/logrus
   version: ba1b36c82c5e05c4f912a88eab0dcd91a171688f
-  repo: https://github.com/sirupsen/logrus
+  repo: https://github.com/Sirupsen/logrus
```
Comment (jackfrancis) [inline glide.lock]: Please remove this change from this PR, we'll manage this in a separate `glide` PR, thanks!

File: glide.yaml [hunk 2]
```diff
@@ -20,7 +20,7 @@ import:
   - autorest/date
 - package: github.com/Sirupsen/logrus
   version: v0.11.5
-  repo: https://github.com/sirupsen/logrus
+  repo: https://github.com/Sirupsen/logrus
```
Comment (jackfrancis) [inline glide.yaml]: ditto

File: pkg/acsengine/const.go [hunk 3]
```diff
@@ -158,6 +158,11 @@ var KubeConfigs = map[string]map[string]string{
 }
 
 const (
+	// Root URL for extensions
+	DefaultExtensionsRootURL = "https://raw.githubusercontent.com/Azure/acs-engine/master/"
```
Comment (jackfrancis) [inline pkg/acsengine/const.go]: Could you move this `const` down to the pre-existing `const` block?

File: glide.lock (hunk 1 above)
Comment (JackQuincy) [inline glide.lock]: I couldn't build this PR without it, so I'll leave it here and remove it after the other Pr goes through

File: pkg/acsengine/const.go (hunk 3 above)
Comment (JackQuincy) [inline pkg/acsengine/const.go]: this file has a bunch of them but these are all urls so sure I can move this one.

**Comment by JackQuincy** [comment]
So there has been a change in kubernetes provision script that isn't compatiable with this design. I'm going to be closing this until I can adjust the design for the new changes.

**Pull request closed without merging**